src/network/p2p.rs

```diff
@@ -48,13 +49,13 @@ use std::iter;
 ///  * [`EventLoop`]: used to process events received from the p2p [`Swarm`] and commands from the p2p [`Client`]
 pub async fn create_components(
 ) -> Result<(Client, impl Stream<Item = Event>, EventLoop), Box<dyn Error>> {
-    let local_keys = identity::Keypair::generate_ed25519();
+    let local_keypair = keypair_util::create_ed25519();
```

Comment:
"create" does not really capture the magic this function is doing, maybe this?

```suggestion
    let local_keypair = keypair_util::initiate_ed25519();
```

Reply:
I had indeed some trouble to find a good name. This is what the function does: load a key from file, if it's not there, create a new one and save that one for future calls. `initiate` makes it sound like it creates a new keypair each time it's called. Would `load_or_create_ed25519` be any better?

Reply:
I agree create is misleading and initiate is vague, so +1 from me for load_or_create